### PR TITLE
Fix batched DiffusionGemma adaptive stopping

### PR DESCRIPTION
**Pull request by kashif**
DiffusionGemma adaptive stopping currently waits for the whole batch to converge, so rows that finish early keep changing while slower rows continue. It also measures confidence from the raw model logits instead of the scheduler-shaped logits used for sampling.

This freezes each row once it is stable and confident, and uses the scheduler logits for the stopping decision. A regression test covers both cases.

Checks:
- `make quality`
- `make fix-copies`
- DiffusionGemma pipeline tests (`11 passed`)
- compiled decoder with static cache smoke test

### CODE REVIEW

File: src/diffusers/pipelines/diffusion_gemma/pipeline_diffusion_gemma.py [hunk 1]
```diff
@@ -380,7 +383,11 @@ def __call__(
                 canvas = scheduler_output.prev_sample
                 # Self-condition on the logits the scheduler sampled from: temperature-shaped for the reference
                 # EntropyBound sampler, the raw denoiser logits for the others.
-                self_conditioning_logits = scheduler_output.pred_logits
+                pred_logits = scheduler_output.pred_logits
+                if finished_denoising.any():
+                    canvas = torch.where(finished_denoising[:, None], argmax_canvas, canvas)
+                    pred_logits = torch.where(finished_denoising[:, None, None], self_conditioning_logits, pred_logits)
```
Comment (dg845) [inline src/diffusers/pipelines/diffusion_gemma/pipeline_diffusion_gemma.py]: Can you check whether freezing the finished rows here has any effect? Claude says that these lines don't affect the pipeline output (although they possibly affect callback outputs).

File: tests/pipelines/diffusion_gemma/test_diffusion_gemma.py [hunk 2]
```diff
@@ -103,6 +104,46 @@ def test_generate(self):
         self.assertEqual(sequences.shape, (1, self.canvas_length))
         self.assertEqual(len(texts), 1)
 
+    def test_adaptive_stopping_freezes_finished_rows_and_uses_scheduler_logits(self):
```
Comment (dg845) [inline tests/pipelines/diffusion_gemma/test_diffusion_gemma.py]: I think it would be better to test each behavior (freezing finished rows, using scheduler logits) in separate tests, rather than having one test for both behaviors.

File: tests/pipelines/diffusion_gemma/test_diffusion_gemma.py [hunk 3]
```diff
@@ -103,6 +104,46 @@ def test_generate(self):
...
+        self.pipe.model.config.get_text_config(decoder=True).vocab_size = vocab_size
+        forward_calls = 0
+
+        def forward(decoder_input_ids, **kwargs):
+            nonlocal forward_calls
+            forward_calls += 1
+            batch_size, canvas_length = decoder_input_ids.shape
+            return SimpleNamespace(logits=torch.zeros(batch_size, canvas_length, vocab_size))
+
+        class DeterministicScheduler:
```
Comment (dg845) [inline tests/pipelines/diffusion_gemma/test_diffusion_gemma.py]: I think using a real scheduler (e.g. a real `EntropyBoundScheduler`, since it shapes the model logits) would be better here.

File: src/diffusers/pipelines/diffusion_gemma/pipeline_diffusion_gemma.py [hunk 4]
```diff
@@ -223,8 +223,9 @@ def __call__(
                 Only used when `confidence_threshold` is set.
             confidence_threshold (`float`, *optional*, defaults to `0.005`):
                 Leave a block's denoising loop early once every example is stable (see `stability_threshold`) and the
```
Comment (dg845) [inline src/diffusers/pipelines/diffusion_gemma/pipeline_diffusion_gemma.py]: nit: the `confidence_threshold` docstring should be updated to reflect the fact that it is now applied per-example. (I think `stability_threshold` may also need an analogous change.)

**Review by dg845 — COMMENTED** [review]
Thanks for the PR! Are the changes (committing finished rows immediately, using scheduler-shaped logits instead of raw denoising model logits) intended to match the DiffusionGemma reference behavior? It's not obvious to me that they are bugfixes rather than modeling changes.

**Comment by kashif** [comment]
thanks @dg845 fixing!

File: tests/pipelines/diffusion_gemma/test_diffusion_gemma.py [hunk 5]
```diff
@@ -103,6 +117,61 @@ def test_generate(self):
...
+
+            def set_timesteps(self, num_inference_steps, device=None):
+                self.timesteps = torch.arange(num_inference_steps, device=device)
+
+            def step(self, model_output, timestep, sample, return_dict=True):
+                del timestep, return_dict
+                return SimpleNamespace(prev_sample=model_output.argmax(dim=-1), pred_logits=model_output)
+
+        self.pipe.model.forward = forward
+        self.pipe.scheduler = DeterministicScheduler()
```
Comment (dg845) [inline tests/pipelines/diffusion_gemma/test_diffusion_gemma.py]:
```suggestion
        self.pipe.scheduler = BlockRefinementScheduler()
```
If we use a real `BlockRefinementScheduler` here, the `test_adaptive_stopping_freezes_finished_rows` test should still pass, so I think we should use a real scheduler over `DeterministicScheduler` (which we can remove after this suggested change), like in `test_adaptive_stopping_uses_scheduler_logits` below.

**Review by dg845 — APPROVED** [review]
Thanks for the changes! Can we also update the `DiffusionGemma` docs to reflect the new per-batch-example behavior? Specifically, I think these lines need to be updated:

https://github.com/huggingface/diffusers/blob/9284607295a09f759aadd65ed08f48b35feea6d9/docs/source/en/api/pipelines/diffusion_gemma.md?plain=1#L148-L152

(For my question about the whether the changes are a bugfix, I believe the answer is that this PR's changes update the `diffusers` implementation to match the `transformers` implementation's batched adaptive stopping behavior.)
